CRM/Core/ClassLoader.php

```diff
@@ -91,6 +91,23 @@ protected function __construct() {
   }
 
   /**
+   * Requires the autoload.php generated by composer
+   *
+   * @return void
+   */
+  protected function requireComposerAutoload() {
+    $civicrm_base_path = dirname(dirname(__DIR__));
```

Comment:
Pretty sure you can use

    $civicrm_base_path = dirname(__FILE__, 2);
    $top_path = dirname(__FILE__, 5);

Would be good to include a comment clarifying what's magic about the 5-th parent directory (and/or use a Drupal method to discover Drupal root, if that's what that is).